test/jdk/com/sun/jdi/JdbOptions.java

```diff
@@ -49,22 +57,32 @@ static String propString(String name, String value) {
         return "prop[" + name + "] = >" + value + "<";
     }
 
-    public static void main(String[] args) {
-        System.out.println(OK_MSG);
-        // print all args
-        List<String> vmArgs = ManagementFactory.getRuntimeMXBean().getInputArguments();
-        for (String s: vmArgs) {
-            System.out.println(argString(s));
-        }
-        // print requested sys.props
-        for (String p: args) {
-            System.out.println(propString(p, System.getProperty(p)));
+    /**
+     * 1st argument is a filename to redirect application output,
+     * the rest are names of the properties to dump.
+     */
+    public static void main(String[] args) throws IOException {
+        String outFile = args[0];
+        try (PrintStream out = new PrintStream(outFile, StandardCharsets.UTF_8)) {
+            out.println(OK_MSG);
+            // print all args
+            List<String> vmArgs = ManagementFactory.getRuntimeMXBean().getInputArguments();
+            for (String s : vmArgs) {
+                out.println(argString(s));
+            }
+            // print requested sys.props (skip 1st arg which is output filename)
+            for (int i=1; i < args.length; i++) {
+                String p = args[i];
+                out.println(propString(p, System.getProperty(p)));
+            }
         }
     }
 }
 
 public class JdbOptions {
-    private static final String targ = JbdOptionsTarg.class.getName();
+    private static final String outFilename = UUID.randomUUID().toString() + ".out";
+    private static final Path outPath = Paths.get(outFilename);
+    private static final String targ = JbdOptionsTarg.class.getName() + " " + outFilename;
```

Comment:
Is there a reason you added `outFilename` to `targ` rather than just add `outFilename` when you pass the argument string to `test()`.

Reply:
I think `targ` is typically the classfile name of the target debuggee, so I think it would be best to keep the output filename separate from it to avoid confusion.